homeassistant/components/command_line/sensor.py

```diff
@@ -89,50 +92,62 @@ async def async_setup_platform(
         value_template.hass = hass
     json_attributes: list[str] | None = sensor_config.get(CONF_JSON_ATTRIBUTES)
     scan_interval: timedelta = sensor_config.get(CONF_SCAN_INTERVAL, SCAN_INTERVAL)
+    state_class: SensorStateClass | None = sensor_config.get(CONF_STATE_CLASS)
     data = CommandSensorData(hass, command, command_timeout)
 
+    trigger_entity_config = {
+        CONF_UNIQUE_ID: unique_id,
+        CONF_NAME: Template(name, hass),
+        CONF_DEVICE_CLASS: sensor_config.get(CONF_DEVICE_CLASS),
+    }
+
     async_add_entities(
         [
             CommandSensor(
                 data,
-                name,
+                trigger_entity_config,
                 unit,
+                state_class,
                 value_template,
                 json_attributes,
-                unique_id,
                 scan_interval,
             )
         ]
     )
 
 
-class CommandSensor(SensorEntity):
+class CommandSensor(ManualTriggerEntity, SensorEntity):
     """Representation of a sensor that is using shell commands."""
 
     _attr_should_poll = False
 
     def __init__(
         self,
         data: CommandSensorData,
-        name: str,
+        config: ConfigType,
         unit_of_measurement: str | None,
+        state_class: SensorStateClass | None,
         value_template: Template | None,
         json_attributes: list[str] | None,
-        unique_id: str | None,
         scan_interval: timedelta,
     ) -> None:
         """Initialize the sensor."""
-        self._attr_name = name
+        super().__init__(self.hass, config)
         self.data = data
         self._attr_extra_state_attributes = {}
         self._json_attributes = json_attributes
         self._attr_native_value = None
         self._value_template = value_template
         self._attr_native_unit_of_measurement = unit_of_measurement
-        self._attr_unique_id = unique_id
+        self._attr_state_class = state_class
         self._scan_interval = scan_interval
         self._process_updates: asyncio.Lock | None = None
 
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
```

Comment:
Why is this necessary ? The default property is already returning this value?

Reply:
`BaseTriggerEntity` implements `extra_state_attributes` property for attributes with templates so it's necessary to define the property here

Reply:
Because we are not working with templates at attributes, we just want to show them as they are set at updating

Reply:
Why was this not part of the other PRs like cover?

Reply:
Only `sensor` implements/is using `extra_state_attributes`.

Reply:
We could also "skip" possible templating on attributes for the `ManualTriggerEntity` and implement `extra_state_attributes` property already on `ManualTriggerEntity` to not have to set it here?
Would not think it would ever be used for templating with this one?

Reply:
Or perhaps we should add some sort of "manual" additions to `extra_state_attributes` to allow for both?
`manual_extra_state_attributes`

Reply:
let's keep it as is. I see now how it's used. It does feel like the way trigger template entities use it makes more sense.